### PR TITLE
feat: agents apis

### PR DESCRIPTION
**Pull request by tve-corti**
## Summary

This pull request adds two new helper methods to the SDK to improve developer experience when working with agent cards and authentication headers.

### 1. Add `agents.getCardUrl()` method

Introduces a new helper method `getCardUrl(agentId: string): Promise<URL>` to the agents client that constructs the URL for retrieving an agent's card JSON file. The method automatically resolves the correct environment-specific base URL and builds the path `/agents/{agentId}/agent-card.json`.

**Implementation:**
- Creates new `CustomAgents` class extending the auto-generated `AgentsClient`
- Integrates custom agents client into `CortiClient`

**Test coverage:** Unit tests for various agent IDs, environments (eu, us, dev-eu), authentication methods (client credentials, bearer token), and URL encoding edge cases.

### 2. Add `getAuthHeaders()` method

Adds a new public method `getAuthHeaders(): Promise<Headers>` to `CortiClient` that returns a Headers object containing the Authorization header with the current access token and the Tenant-Name header. This simplifies header construction for developers making custom API requests.

**Implementation:**
- Retrieves authentication request from the configured auth provider
- Combines Authorization header with tenant name from client options

**Test coverage:** Comprehensive tests covering client credentials, bearer tokens, async token retrieval, error propagation, and header structure validation.

## Files Modified

- CortiClient.ts
- CustomAgents.ts (new)
- agents.getCardUrl.integration.ts (new)
- cortiClient.getAuthHeaders.integration.ts (new)

### CODE REVIEW

File: src/custom/agents/CustomAgents.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * All the patches marked with `// Patch: ...` comments.
+ */
+
+import { AgentsClient  } from "../../api/resources/agents/client/Client.js";
+import * as core from "../../core/index.js";
+
+export class CustomAgents extends AgentsClient {
+    /**
+     * Patch: added helper method to get agent card URL
```
Comment (markitosha) [inline src/custom/agents/CustomAgents.ts]: I actually followed your recommendation to not use "Patch" comments way too much (there was a task about it), and now I'm using these comments only for complete overrides (when we had to copy-paste the whole file and patch something there). For custom logic on top of the generated code they indeed don't make any sense

**Review by markitosha — COMMENTED** [review]
LGTM 👍 
Only one comment about comment 😅 (and you have to fix linter anyway)
Ping me if you need instructions how to "release" a version (I believe I didn't update docs yet)!

File: src/custom/agents/CustomAgents.ts (hunk 1 above)
Comment (tve-corti) [reply src/custom/agents/CustomAgents.ts]: So here's what happened: I based this on a 6 month old branch, realised my mistake at 15:45 and then hastily cherry picked and resolved conflicts in 10 minutes so I coul get a review. The result was a pile of garbage. I've tidied things up properly now.

**Review by markitosha — APPROVED** [review]
